core/src/module/wit_instance.rs

```diff
@@ -136,22 +133,22 @@ impl ITInstance {
             .collect::<MResult<HashMap<_, _>>>()
     }
 
-    fn extract_memories(wasmer_instance: &WasmerInstance) -> Vec<WITMemory> {
-        use wasmer_core::export::Export::Memory;
+    fn extract_memories(
+        wasm_instance: &<WB as WasmBackend>::Instance,
+        store: &mut <WB as WasmBackend>::Store,
+    ) -> Vec<<WB as WasmBackend>::Memory> {
+        use marine_wasm_backend_traits::Export::Memory;
 
-        let mut memories = wasmer_instance
-            .exports()
+        let mut memories = wasm_instance
+            .export_iter(store.as_context_mut())
             .filter_map(|(_, export)| match export {
-                Memory(memory) => Some(WITMemory(memory)),
+                Memory(memory) => Some(memory),
                 _ => None,
             })
             .collect::<Vec<_>>();
 
-        if let Some(Memory(memory)) = wasmer_instance
-            .import_object
-            .maybe_with_namespace("env", |env| env.get_export("memory"))
-        {
-            memories.push(WITMemory(memory));
+        if let Ok(memory) = wasm_instance.get_memory(store, "memory") {
```

Comment:
Is the `"memory"` a some kind of standard name?

Reply:
Yes. I changed it to a named constant, so it wont confuse anyone anymore.